src/logging.c

```diff
@@ -629,8 +706,13 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
   if (chars_to_copy > RCUTILS_LOGGING_MAX_OUTPUT_FORMAT_LEN - 1) {
     chars_to_copy = RCUTILS_LOGGING_MAX_OUTPUT_FORMAT_LEN - 1;
   }
-  memcpy(g_rcutils_logging_output_format_string, output_format, chars_to_copy);
-  g_rcutils_logging_output_format_string[chars_to_copy] = '\0';
+
+  // decode the escape characters
+  rcutils_ret_t decode_ret = decode_escape_characters(output_format, chars_to_copy);
```

Comment:
OK. Since the `chars_to_copy` was not used in the original function below, move it into the new function `create_format_string`.

Updated in https://github.com/ros2/rcutils/commit/2226ca4aa698ce0fdf6afc4f72768077a6ac11e4